File: app/src/main/java/net/chris/demo/guardian/ui/recycler/SectionListItem.java
package net.chris.demo.guardian.ui.recycler;

public class SectionListItem {

    private String id;

    private String sectionName;

    private boolean isSelected;

    private SectionListItem(String id, String sectionName, boolean isSelected) {
        this.id = id;
        this.sectionName = sectionName;
        this.isSelected = isSelected;
    }

    public String getId() {
        return id;
    }

    public String getSectionName() {
        return sectionName;
    }

    public boolean isSelected() {
        return isSelected;
    }

    public void setId(String id) {
        this.id = id;
    }

    public void setSectionName(String sectionName) {
        this.sectionName = sectionName;
    }

    public void setSelected(boolean selected) {
        isSelected = selected;
    }

    public static class Builder {
        private String id;
        private String sectionName;
        private boolean isSelected;

        public Builder setId(String id) {
            this.id = id;
            return this;
        }

        public Builder setSectionName(String sectionName) {
            this.sectionName = sectionName;
            return this;
        }

        public Builder setSelected(boolean selected) {
            this.isSelected = selected;
            return this;
        }

        public SectionListItem build() {
            return new SectionListItem(id, sectionName, isSelected);
        }
    }
}

File: app/src/main/kotlin/net/chris/demo/guardian/ui/recycler/SectionListItem.kt
package net.chris.demo.guardian.ui.recycler

class SectionListItem private constructor(
        var id: String?,
        var sectionName: String?,
        var isSelected: Boolean
) {

    class Builder {
        private var id: String? = null
        private var sectionName: String? = null
        private var isSelected: Boolean = false

        fun setId(id: String): Builder {
            this.id = id
            return this
        }

        fun setSectionName(sectionName: String): Builder {
            this.sectionName = sectionName
            return this
        }

        fun setSelected(selected: Boolean): Builder {
            this.isSelected = selected
            return this
        }

        fun build(): SectionListItem {
            return SectionListItem(id, sectionName, isSelected)
        }
    }
}
